tests/test_constraints.py

```diff
@@ -754,6 +754,22 @@ class Ex(msgspec.Struct):
             with pytest.raises(msgspec.ValidationError, match=err_msg):
                 dec.decode(proto.encode(Ex(x)))
 
+    def test_key_pattern(self, proto):
+        class Ex(msgspec.Struct):
+            x: Dict[Annotated[str, Meta(pattern="[a-z]+")], int]
+
+        dec = proto.Decoder(Ex)
+
+        x = {"a": 1}
+        assert dec.decode(proto.encode(Ex(x))).x == {"a": 1}
+
+        x = {"1": 0}
+        err_msg = re.escape(
+            r"Expected `str` matching regex '[a-z]+' - at `key` in `$.x`"
+        )
+        with pytest.raises(msgspec.ValidationError, match=err_msg):
```

Comment:
@jcrist any thoughts here? this does _not_ raise on the `msgpack` backend.

Reply:
Thanks for raising this (and apologies for the delayed response - I'm currently out on leave). This specific issue should be resolved in #577.